x-pack/legacy/plugins/ml/server/models/data_frame/transforms.ts

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { callWithRequestType } from '../../../common/types/kibana';
+import {
+  DATA_FRAME_TRANSFORM_STATE,
+  DataFrameTransformEndpointData,
+} from '../../../public/data_frame/pages/transform_management/components/transform_list/common';
+
+export function transformServiceProvider(callWithRequest: callWithRequestType) {
+  async function deleteTransform(transformId: string) {
+    return callWithRequest('ml.deleteDataFrameTransform', { transformId });
+  }
+
+  async function stopTransform(options: {
```

Comment:
`stopTransform` and `startTransform` take a very similar `options` parameter, could the types be combined in a shared interface?

Reply:
Since these are very small functions I found keeping them separate felt okay right now as we may change/add the options we pass to each and it feels much cleaner keeping them isolated than in the future having the potential of many if else's or ternary operators in a single function. 🤔 

Having said that, I'm happy to update it now if everyone feels it's the best options. Otherwise, I'd like to wait and see what we end up adding to Data Frames and update it in the future if needed.